accord-core/src/main/java/accord/local/CommandStores.java

```diff
@@ -693,7 +758,15 @@ protected <O> AsyncChain<O> mapReduce(PreLoadContext context, MapReduce<? super
     public synchronized Supplier<EpochReady> updateTopology(Node node, Topology newTopology, boolean startSync)
     {
         TopologyUpdate update = updateTopology(node, current, newTopology, startSync);
-        current = update.snapshot;
+        // TODO (review/discussion): an alternative to this would be to store all previous topology updates (or base compacted image + updates),
+        //      and compact them into a single record. This can be done either in this patch or in a follow-up.
+        if (update.snapshot != current)
+        {
+            AsyncResults.SettableResult<Void> flush = new AsyncResults.SettableResult<>();
```

Comment:
as discussed, let's chain the EpochReady AsyncResult fields to complete after this flush